tests/MongoDB.Driver.Tests/Specifications/server-selection/ServerSelectionTests.cs

```diff
@@ -0,0 +1,93 @@
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.server_selection
+{
+    public class ServerSelectionTest
+    {
+        [Theory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find") || e.CommandName.Equals("$query"));
+            using (var subject = CreateDisposableClient(eventCapturer, ReadPreference.PrimaryPreferred))
+            {
+                var database = subject.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+                var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+
+                if (async)
+                {
+                    collection.FindAsync("{ x : 2 }").GetAwaiter().GetResult();
+                }
+                else
+                {
+                    collection.FindSync("{ x : 2 }");
+                }
+
+                BsonDocument resultCommand;
+                var serverVersion = subject.Cluster.Description.Servers[0].Version;
+
+                if (subject.Cluster.Description.Type == ClusterType.Standalone)
+                {
+                    resultCommand = ((CommandStartedEvent)eventCapturer.Events[0]).Command;
+
+                    resultCommand.Contains("readPreference").Should().BeFalse();
+                }
+                else if (subject.Cluster.Description.Type == ClusterType.Sharded &&
+                         serverVersion < Feature.CommandMessage.FirstSupportedVersion)
+                {
+                    resultCommand = ((CommandStartedEvent)eventCapturer.Events[0]).Command;
+                    if (((CommandStartedEvent) eventCapturer.Events[0]).CommandName.Equals("$query"))
+                    {
+                        resultCommand.Contains("$readPreference").Should().BeTrue();
+                    }
+                    else if (((CommandStartedEvent) eventCapturer.Events[0]).CommandName.Equals("find"))
+                    {
+                        resultCommand.Contains("readPreference").Should().BeTrue();
+                    }
+                }
+                else if (serverVersion >= Feature.CommandMessage.FirstSupportedVersion)
+                {
+                    resultCommand = ((CommandStartedEvent)eventCapturer.Events[0]).Command;
+                    resultCommand.Contains("$readPreference").Should().BeTrue();
+                }
```

Comment:
I think it might be clearer to combine the last two clauses as follows:

    else if (subject.Cluster.Description.Type == ClusterType.Sharded)
    {
        if (serverVersion < Feature.CommandMessage.FirstSupportedVersion)
        {
            // asserts for older server versions
        }
        else
        {
            // asserts for newer server versions
        }
    }

But... why do we even need different asserts?

Maybe we just need to figure out why we are seeing `$query` as the command name (that's *clearly* NOT a command name) and fix that.

Reply:
The appearance of `$query` confused me as well, and I wrote the extra asserts to account for it. But I agree, I think it may be more useful to determine why it's there.

Reply:
Let us know what you find out about why you are seeing the unexpected `$query`.

Reply:
See conversation below.